src/dp_flow.c

```diff
@@ -123,6 +124,7 @@ static __rte_always_inline void dp_mark_vnf_type(struct dp_flow *df, const struc
 	struct snat_data *s_data;
 	struct dp_ip_address pfx_ip;
 
+	// what if key->l3_type == RTE_ETHER_TYPE_IPV6? (see caller)
 	memset(&pfx_ip, 0, sizeof(pfx_ip));
 	pfx_ip.ip_type = RTE_ETHER_TYPE_IPV4;
 	pfx_ip.ipv4 = key->l3_src.ip4;
```

Comment:
This is again just for LB prefix lookup, so maybe just change the condition below to check `l3_type` and only then pull the IPv4 out?

Reply:
Yes, the suggestion is fine.